### ISSUE TITLE
mention authorea at the thing

### ISSUE THREAD

**Issue by silky** [OpenScienceWorkshops/OpenScienceWorkshops.github.io#50]
because it's awesome.

**Comment by silky** [comment]
i'm not so sure now.

authorea is a bit broken at the moment, see (issues reported by me only; more by other people that are also relevant):
- https://authorea.com/issues/198
- https://authorea.com/issues/199
- https://authorea.com/issues/202 (though the soln here is obviously just github for now)
- https://authorea.com/issues/203 (kind of serious)

the last one is really annoying. it means i can't use it at all; and i'll not try and do my thesis on it, unless they somehow fix that with some urgency.

glancing through the other issues though, they don't seem to fix things super-quickly. (for example - https://authorea.com/issues/39 this could be pretty easy to do, but has been open for 2 months; otoh i guess it could be very complicated or just not as important as other things.)

i'm wondering if maybe it's best to talk about something else on the day, something that "just works"? my reluctance is that we'll have to say "here's this thing, it could be great but it isn't yet!". that's kind of an annoying statement to have to make to a bunch of non-experts - what's the point? tell us when we can use it.

there's writeLaTeX - https://www.writelatex.com/ or shareLaTeX - https://www.sharelatex.com/

i've not used either and i dont' know what the difference is. gerry has though.

**Comment by andreabedini** [comment]
Yeah, I have been experiencing some issue too, I wrote to Alberto & Nathan, they said they are looking into fixing things.

**Comment by silky** [comment]
i guess we'll see how it goes on a few weeks and decide then. i still have high hopes for it.

**Comment by andreabedini** [comment]
yeah me too, writeLaTeX and shareLaTeX are “tools” while Authorea has a
vision of open science.

**Comment by silky** [comment]
it's been a while now, and still no meaningful update.

i'm torn. i don't think i could use authorea for anything at the moment. i certainly can't use it for notes from math classes. i can't use it for my thesis, i don't think, because i'm not confident enough in the pandoc workflow for TeX (what they use) for that.

i guess i could give that a go, and see if i can figure out why they can't even get xymatrix working. it shouldn't be hard, then maybe check in with them in a week or so.

**Comment by andreabedini** [comment]
I'm torn too. We should really have a chat with them :-/

**Comment by silky** [comment]
i just checked; it's unspurisingly trivial to do whatever you want with pandoc, as expected.

you build a pdf from TeX with pandoc like so:

```
pandoc -N --template=mytemplate.tex --variable fontsize=12pt --variable version=1.10 README --latex-engine=xelatex --toc -o example14.pdf
```

so it takes a template, and variables, and other options.

variables are whatever you want.

the 'mytemplate.tex' can be found here - http://johnmacfarlane.net/pandoc/demo/mytemplate.tex - and it's just a raw TeX file, with specific tags to denote sections.

in other words, it can be whatever you want. so i got it to generate xymatrix output happily.

it would remain only to change my build process here - https://github.com/silky/alg-top-notes/ - to use pandoc instead of (or as well as) latexmk.

but perhaps this is a bit of a side issue anyway. i'll get an email thread started.

**Comment by silky** [comment]
yeah, i've worked out why the math notes thing i was trying to work on didn't work on their site - they just simply don't have the `amsthm` and `xy` environments.

this explains why the rendered pdf doesn't work.

but as for the document _itself_ being basically unreadable - i don't think there is an easy solution. partly it's fixed by using xyjax (i can't see them doing this soon). as for actually rendering the custom theorem environments; i'd be tempted to claim that that verges on flat-out impossible (by "impossible" i don't mean impossible, i just mean it would take parsing TeX itself to do it right.)

pandoc doesn't do it (say when converting text -> md), and in general figuring out what the latex definitions should do is something that i can't imagine any sane system actually doing. maybe pandoc has some advice on dealing with custom TeX environments.

**Comment by andreabedini** [comment]
I know, my lecture notes are done with markdown + panadoc ;)

**Comment by silky** [comment]
haha! nice!

actually that's a good point. maybe this authorea thing can be good for lecture notes ...

**Comment by silky** [comment]
andrea did this.
